### PR TITLE
Feat add crossconnect

### PR DESCRIPTION
**Pull request by Syndlex**
### Description of your changes

This PR adds: Crossconnect as crd

## Checklist

I have:

- [x] Add PR name as appropriate (e.g. `feat`/`fix`/`doc`/`test`/`refactor`)
- [x] Run `make reviewable` and `make crds.clean` to ensure the PR is ready for review
- [ ] Add or update tests (if applicable) Nope just copied everything from datacenter
- [x] Add or update Documentation using `make docs.update` (if applicable)
- [x] Update `docs/CHANGELOG.md` file (label: `upcoming release`)
- [ ] Check Sonar Cloud Scan
- [ ] Update Github or Jira Issue

### CODE REVIEW

File: examples/ionoscloud/compute/privatecrossconnect.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Use this file to create a Datacenter CR.
+# Required values for creating a Datacenter CR are:
+# - Location.
+apiVersion: compute.ionoscloud.crossplane.io/v1alpha1
+kind: PrivateCrossConnect
+metadata:
+  name: examplePcc
```
Comment (cristiGuranIonos) [inline examples/ionoscloud/compute/privatecrossconnect.yaml:7]: lowercase, just set it to example, or examplepcc

File: examples/ionoscloud/compute/privatecrossconnect.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Use this file to create a Datacenter CR.
```
Comment (cristiGuranIonos) [inline examples/ionoscloud/compute/privatecrossconnect.yaml:1]: should be replaced with pcc description

File: package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+spec:
+  group: compute.ionoscloud.crossplane.io
+  names:
+    categories:
+    - crossplane
+    - managed
+    - ionoscloud
+    kind: PrivateCrossConnect
+    listKind: PrivateCrossConnectList
+    plural: privatecrossconnects
```
Comment (cristiGuranIonos) [inline package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml:17]: better to have these as pcc and pccs

**Comment by cristiGuranIonos** [comment]
thanks for the CR, I think it needs to be added to integration tests. WIll try to create some pccs to see exactly how it works.

File: package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml (hunk 3 above)
Comment (Syndlex) [reply package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml:17]: this is auto generated. Where do i have to change those?

**Comment by Syndlex** [comment]
... I currently have a ticket open for getting Information how Pccs work with k8s. Internal: #:207140121

File: package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml (hunk 3 above)
Comment (cristiGuranIonos) [reply package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml]: sadly, I saw that basically change the name of the structures.

**Comment by cristiGuranIonos** [comment]
One more small thing, in `internal/controller/ionoscloud.go` pcc.Setup needs to be called like for the other resources. Thanks!

**Comment by Syndlex** [comment]
@cristiGuranIonos thanks for looking at it. :) 

There where some more compiling Problems today... strange i executed `make reviewable` after the renaming stuff.

File: package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml (hunk 3 above)
Comment (Syndlex) [reply package/crds/compute.ionoscloud.crossplane.io_privatecrossconnects.yaml]: jup :/

I dont really like acronyms. You have to decide what todo. :)

File: examples/ionoscloud/compute/lan.yaml [hunk 4]
```diff
@@ -11,7 +11,9 @@ managementPolicies:
 spec:
   forProvider:
     name: exampleLan
-    #    pcc:
+    #pcc:
+    #  PrivateCrossConnectIdRef:
+    #    name: examplePcc
```
Comment (cristiGuranIonos) [inline examples/ionoscloud/compute/lan.yaml:16]: would have been good here also. will change it after, when I add tests

**Comment by cristiGuranIonos** [comment]
> @cristiGuranIonos thanks for looking at it. :)
> 
> There where some more compiling Problems today... strange i executed `make reviewable` after the renaming stuff.

From what I saw the lint passed. Sometimes the code generation doesn't work as good.

**Comment by cristiGuranIonos** [comment]
One small thing also, this is technically a small breaking change for lans so have to discuss about how we will release

**Comment by Syndlex** [comment]
Would be great to have this sone. :)  

I would like to use this but when it takes to long i can not convince management to wait.

**Review by cristiGuranIonos — APPROVED** [review]

**Pull request merged**